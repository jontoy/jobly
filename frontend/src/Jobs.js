import React, {useState, useEffect} from 'react';
import Searchbox from './Searchbox';
import JoblyApi from './JoblyApi';
import JobsList from './JobsList';
import SelectField from './SelectField';
import Pagination from './Pagination';
import usePagination from './hooks/usePagination';
import useApply from './hooks/useApply';
const Jobs = () => {
    const [isLoading, setIsLoading] = useState(true);
    const [jobs, setJobs, page, setPage, resultsPerPage, setResultsPerPage, goBack, goForward] = usePagination(20);
    const [searchTerm, setSearchTerm] = useState('');
    const [apply, unapply] = useApply();


    useEffect(() => {
      async function getJobs() {
        let allCompanies = await JoblyApi.getJobs({search:searchTerm});
        setJobs(allCompanies);
      }
      getJobs();
      setIsLoading(false);
      setPage(1);
    }, [searchTerm, setJobs, setPage]);

    if (isLoading) {
      return <p>Loading &hellip;</p>;
    }
    return (
        <div className="Jobs container">
            <Searchbox filterResults={setSearchTerm}/>
            <JobsList jobs={jobs.slice(resultsPerPage*(page-1), resultsPerPage*page)} apply={apply} unapply={unapply}/>
            <Pagination page={page} goBack={goBack} goForward={goForward}/>
            <SelectField headDesc='Show me ' tailDesc=' results per page' options={[10,20,50,100]} value={resultsPerPage} action={setResultsPerPage} subaction={setPage} />
        </div>
    )
}

export default Jobs;