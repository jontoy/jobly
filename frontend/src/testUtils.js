import React from "react";
import UserContext from "./userContext";

const demoUser = {
  username: "test",
  first_name: "test",
  last_name: "mctest",
  email: "test@test.net",
  photo_url: null,
  jobs:[]
};

const UserProvider = ({ children, currentUser = demoUser, appliedIds = [] }) => (
  <UserContext.Provider value={{currentUser, appliedIds}}>
    {children}
  </UserContext.Provider>
);

export { UserProvider };
