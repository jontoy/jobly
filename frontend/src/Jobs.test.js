import React from 'react';
import { render } from '@testing-library/react';
import Jobs from './Jobs';
import {UserProvider} from './testUtils';

it('renders without crashing', () => {
    render(
    <UserProvider>
        <Jobs />
    </UserProvider>
    );
});
  
it('matches snapshot when user is logged in', () => {
      const {asFragment} = render(
        <UserProvider>
            <Jobs />
        </UserProvider>
        );
    expect(asFragment()).toMatchSnapshot();
});