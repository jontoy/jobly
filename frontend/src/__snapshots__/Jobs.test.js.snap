// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot when user is logged in 1`] = `
<DocumentFragment>
  <div
    class="Jobs container"
  >
    <form
      class="Searchbox form-inline mt-3"
    >
      <input
        class="form-control form-control-lg flex-grow-1"
        name="search"
        placeholder="Search..."
        type="search"
        value=""
      />
    </form>
    <ul
      class="JobsList list-unstyled"
    />
    <nav
      aria-label="Pagination"
      class="Pagination"
    >
      <ul
        class="pagination"
      >
        <li
          class="page-item"
        >
          <button
            class="page-link"
          >
            Previous
          </button>
        </li>
        <li
          class="page-item"
        >
          <button
            class="page-link"
          >
            1
          </button>
        </li>
        <li
          class="page-item"
        >
          <button
            class="page-link"
          >
            Next
          </button>
        </li>
      </ul>
    </nav>
    <div
      class="SelectField"
    >
      Show me 
      <select>
        <option
          value="10"
        >
          10
        </option>
        <option
          value="20"
        >
          20
        </option>
        <option
          value="50"
        >
          50
        </option>
        <option
          value="100"
        >
          100
        </option>
      </select>
       results per page
    </div>
  </div>
</DocumentFragment>
`;
